spec/test_app_templates/lib/generators/test_app_generator.rb

```diff
@@ -28,11 +28,28 @@ def inject_css
   end
 
   def inject_javascript
-    insert_into_file 'app/assets/javascripts/application.js', after: '//= require_tree .' do
-      %(
-        //= require jquery
-        //= require browse_everything
-      )
+    if Rails.version =~ /^6\./
+      copy_file 'app/assets/javascripts/application.js', 'app/assets/javascripts/application.js'
+
+      # Does this means that Webpacker becomes a hard-dependency?
```

Comment:
Nope, this has nothing to do with webpacker. In fact, it's the opposite -- what this line of code is doing is making sure that _Sprockets_ will still handle Javascript, even if you are using Sprockets 4. 

Since:

* Sprockets 4 (unlike sprockets 3) will use this `app/assets/config/mannifest.js` file 
* Rails 6 (unlike Rails 5) will not include the `link_directory ../javascripts .js` line in the generated `app/assets/config/manifest.js` file, since Rails 6 thinks you shouldn't use Sprockets for JS. 

So what you're doing is making it so Sprockets will still do JS in Rails 6.0.1 with Sprockets 4 -- nothing to do with webpacker at all. 

So at least, you probably want to remove/change this comment. 

But also, since this is a generator -- you probably want to document where (README?) that the browse-everything installer under Rails 6 is going to re-enable Sprockets handling of JS, and add a line to your application.html.erb layout that generates a <script> tag for a sprockets-compiled application.js in addition to the existing webpacker-compiled one. 

Which is not ideal -- but is better than browse-everything not working at all, it's what we've got for Rails 6 right now.